### DISCUSSION TITLE
Strongswan monitoring

### DISCUSSION THREAD

**Discussion by ptchuba** [strongswan/strongswan#1886] [Q&A]
I apologize if this has been discussed before but I could not find anything solid on the topic. I'll like to know what tools/scripts you use to monitor your strongswan environment. I have several strongswan boxes and will like to monitor/graph things like number of users connected, number of SAs, Child SAs, tunnel RTT, Tunnel packet loss, Data transferred/Bandwidth usage etc. SNMP would be ideal but I know Strongswan doesn't support that. 

Any recommendations would be greatly appreciated.

Thnanks!!

**Comment by tobiasbrunner** [comment]
> number of users connected, number of SAs, Child SAs, tunnel RTT, Tunnel packet loss, Data transferred/Bandwidth usage

Note that some of these stats are specific to IKE (which strongSwan handles) and others to IPsec (which the kernel handles, and while strongSwan can be queried for some of the data, that might not be complete due to rekeyings etc. and can also be retrieved directly from the kernel via `ip -s xfrm ...`), so there won't be a single source for all the information.

You could look into [_vici_ plugin](https://docs.strongswan.org/docs/5.9/plugins/vici.html), the [_counters_ plugin](https://docs.strongswan.org/docs/5.9/plugins/counters.html), maybe the [_error-notify_ plugin](https://docs.strongswan.org/docs/5.9/plugins/error-notify.html) and possibly [RADIUS accounting](https://docs.strongswan.org/docs/5.9/plugins/eap-radius.html#_accounting).

For some (e.g. RTT or packet loss) you might have to script your own probes that send traffic via tunnels.

As to what people actually do in practice, I've no idea. So maybe some community members have suggestions.
